cmds/cluster/storestats/main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/option"
+)
+
+const (
+	lifecycleConfig = `{
+		"rule": [
+			{
+				"action": {
+					"type": "Delete"
+				},
+				"condition": {
+					"age": 1
+				}
+			}
+		]
+	}`
+)
+
+func main() {
+	key := flag.String("key", "key.json", "google service account key")
+	proj := flag.String("project", "hpc-benchmarking-sandbox", "project")
+	bucketName := flag.String("bucket", "", "bucket name")
+	flag.Parse()
+
+	if len(*bucketName) == 0 {
+		*bucketName = "snapcheck" + strings.ReplaceAll(strings.ToLower(time.Now().Format(time.RFC3339)), ":", "-")
+	}
+	ctx := context.Background()
+
+	// Initialize Google Cloud Storage client
+	client, err := storage.NewClient(ctx, option.WithCredentialsFile(*key))
+	if err != nil {
+		log.Fatalf("Failed to create client: %v", err)
+	}
+	defer client.Close()
+
+	// Create or get the bucket
+	fmt.Printf("gs://%s/%s", *proj, *bucketName)
+	bucket := client.Bucket(*proj)
+
+	if err := bucket.Create(ctx, *proj, nil); err != nil { // && err != storage.ErrBucketExists {
+		gerr, ok := err.(*googleapi.Error)
+		if ok && gerr.Code != 409 {
+			log.Fatalf("create bucket: %v", err)
+		}
+		if !ok {
+			log.Fatalf("create bucket: unknown error type %T:%v", err, err)
+		}
+	}
+
+	// Upload a JSON file to the bucket
+	if err := uploadJSON(ctx, bucket, os.Stdin, *bucketName); err != nil {
+		log.Fatalf("uploading to %q:%v", *bucketName, err)
+	}
+
+	// Apply lifecycle policy
+	if err := setBucketLifecyclePolicy(ctx, bucket); err != nil {
+		log.Fatalf("setting life cycle:%v", err)
+	}
+}
```

Comment:
This implementation creates hard times to implement any testing I think. Some code can be placed in a run function, for easier, better testing.